CHANGELOG.md

```diff
@@ -9,6 +9,89 @@ This is a high level and scarce summary of the changes between releases.
+  number of known failures.  (#2770)
+- The reporting of command results failed if the current working
+  directory was removed (e.g., after an unsuccessful `install`). (#2788)
+- When installing into an existing empty directory, `datalad install`
+  removed the directory after a failed clone.  (#2788)
+- `datalad run` incorrectly handled inputs and outputs for paths with
+  spaces and other characters that require shell escaping.  (#2798)
+- Globbing inputs and outputs for `datalad run` didn't work correctly
+  if a subdataset wasn't installed.  (#2796)
+
+### Enhancements and new features
+
+- The documentation on how to format commands for `datalad run` has
+  been improved.  (#2703)
+- The method for determining the current working directory on Windows
+  has been improved.  (#2707)
+- `datalad --version` now simply shows the version without the
+  license.  (#2733)
+- `datalad export-archive` learned to export under an existing
+  directory via its `--filename` option.  (#2723)
+- `datalad export-to-figshare` now generates the zip archive in the
+  root of the dataset unless `--filename` is specified.  (#2723)
+- After importing `datalad.api`, `help(datalad.api)` (or
+  `datalad.api?` in IPython) now shows a summary of the available
+  DataLad commands.  (#2728)
+- Support for using `datalad` from IPython has been improved.  (#2722)
+- `datalad wtf` now returns structured data and reports the version of
+  each extension.  (#2741)
+- Metadata: changes to egrep mode search  (#2735)
+  - Queries in egrep mode are now case-sensitive when the query
+    contains any uppercase letters and are case-insensitive otherwise.
+    The new mode egrepcs can be used to perform a case-sensitive query
+    with all lower-case letters.
+  - Search can now be limited to a specific key.
+  - Multiple queries (list of expressions) are evaluated using AND to
+    determine whether something is a hit.
+  - A single multi-field query (e.g., `pa*:findme`) is a hit, when any
+    matching field matches the query.
+  - All matching key/value combinations across all (multi-field)
+    queries are reported in the query_matched result field.
+  - egrep mode now shows all hits rather than limiting the results to
+    the top 20 hits.
+- The internal handling of gitattributes information has been
+  improved.  A user-visible consequence is that `datalad create
+  --force` no longer duplicates existing attributes.  (#2744)
+- The "annex" metadata extractor can now be used even when no content
+  is present.  (#2724)
+- The `add_url_to_file` method (called by commands like `datalad
+  download-url` and `datalad add-archive-content`) learned how to
+  display a progress bar.  (#2738)
+- Anonymous access is now supported for S3 and other downloaders.  (#2708)
+- A new interface is available to ease setting up new providers.  (#2708)
+- More codepaths now try multiple times when removing a file to deal
+  with latency and locking issues on Windows.  (#2795)
```

Comment:
WOW! Thank you for doing it!

We might want to position some ahead of the others (e.g. the last ones on providers and anonymous access are imho more important than probably majority of other enhancements) and provide overall summary - major improvements in handling direct and v6 modes and Windows.